### PR TITLE
Adjusted Featured Collection placeholders to work with any number of desktop columns

### PR DESCRIPTION
**Pull request by tyleralsbury**
### PR Summary: 

A quick PR to make the Featured Collection section's empty state match the number of columns on desktop.

### Why are these changes introduced?

A note by @melissaperreault regarding the empty state rendering incorrectly when there are 5 desktop columns.

### What approach did you take?

Changed the loop to use the columns setting.  Note that there are only 4 `product-apparel` placeholders, so I also needed to cycle back to `1` when we got to the 5th.

### Other considerations

Would be nice to have 1 more placeholder to get rid of the nasty `assign` code.

### Decision log

| # | Decision | Alternatives | Rationale | Downsides |
|---|---|---|---|---|
| 1 |   |   |   |   |

### Visual impact on existing themes

Nothing for themes that aren't using Featured Collection in its empty state with exactly 5 columns.

### Testing steps/scenarios
- [ ] Add a Featured Collection section
- [ ] Without setting an actual collection, set the number of columns to 5.
- [ ] Notice that there are now 5 cards shown.

### Demo links

- [Store](https://os2-demo.myshopify.com/?_ab=0&_fd=0&_sc=1&preview_theme_id=162734440470)
- [Editor](https://admin.shopify.com/store/os2-demo/themes/162734440470/editor)

### Checklist
- [ ] Added PR summary for [release notes](https://themes.shopify.com/themes/dawn/styles/default#ReleaseNotes)
- [ ] Requested review from UX (Only for changes that are affecting the experience or perceivable visual details)
- [ ] Created a ticket for the [help.shopify.com](https://help.shopify.com) documentation team about updates to theme settings. (Internal-only task)
- [ ] Followed [theme code principles](https://github.com/Shopify/dawn/blob/main/.github/CONTRIBUTING.md#theme-code-principles)
- [ ] Linted with [Theme Check](https://github.com/Shopify/theme-check)
- [ ] Tested on [mobile](https://shopify.dev/themes/store/requirements#mobile-browser-requirements)
- [ ] Tested on [multiple browsers](https://shopify.dev/themes/store/requirements#desktop-browser-requirements)
- [ ] Tested for [accessibility](https://shopify.dev/themes/best-practices/accessibility)

### CODE REVIEW

**Review by sofiamatulis — COMMENTED** [review]
Small nitpick 👍 The one odd thing is that as you add more columns, the columns are added to the left and not the right

https://screenshot.click/09-08-osshz-aup2r.mp4

File: sections/featured-collection.liquid [hunk 1]
```diff
@@ -98,15 +98,21 @@
...
               {% if settings.animations_reveal_on_scroll %}
                 data-cascade
                 style="--animation-order: {{ forloop.index }};"
               {% endif %}
             >
-              {%- assign placeholder_image = 'product-apparel-' | append: forloop.rindex -%}
+              {% liquid
+                assign idx = forloop.rindex
+                if idx == 5
+                  assign idx = 1
```
[the remark this answers is not in the document]
Comment (sofiamatulis) [reply sections/featured-collection.liquid:112]: It was just a nitpick on the naming, but just wanted to point it out just incase 👍

**Review by eugenekasimov — APPROVED** [review]
🚀

**Review by melissaperreault — APPROVED** [review]
LGTM! 🚢 Thank you!

**Pull request merged**